backend/infrahub/events/models.py

```diff
@@ -100,3 +101,233 @@ def get_related(self) -> list[dict[str, str]]:
+    properties: list[Property] = Field(default_factory=list, description="The properties of this relationship")
+
+
+class CreatedRelationshipCardinalityOne(Relationship):
+    name: str = Field(..., description="The name of the relationship")
+
+    @computed_field
+    def cardinality(self) -> str:
+        return "one"
+
+
+class CreatedRelationshipCardinalityMany(BaseModel):
+    name: str = Field(..., description="The name of the relationship")
+    peers: list[Relationship] = Field(
+        default_factory=list, description="The peers defined during the creation of this relationship"
+    )
+
+    @computed_field
+    def cardinality(self) -> str:
+        return "many"
+
+
+class UpdatedRelationshipCardinalityOne(BaseModel):
+    name: str = Field(..., description="The name of the relationship")
+    peer_id_previous: str | None = Field(..., description="The previous peer of this relationship")
+    peer_kind_previous: str | None = Field(..., description="The node kind of the previous peer")
+    peer_id_current: str | None = Field(..., description="The current peer of this relationship")
+    peer_kind_current: str | None = Field(..., description="The node kind of the current peer")
+    properties: list[UpdatedProperty] = Field(
+        default_factory=list, description="Changes to properties of this relationship if any were made"
+    )
+
+    @computed_field
+    def cardinality(self) -> str:
+        return "one"
+
+    @computed_field
+    def peer_status(self) -> UpdateStatus:
+        """Indicate how the peer was changed during this update"""
+        if self.peer_id_previous == self.peer_id_current:
+            return UpdateStatus.UNMODIFIED
+        if self.peer_id_previous and not self.peer_id_current:
+            return UpdateStatus.REMOVED
+        if self.peer_id_current and not self.peer_id_previous:
+            return UpdateStatus.ADDED
+
+        return UpdateStatus.UPDATED
+
+
+class UpdatedRelationshipPeer(BaseModel):
+    peer_id: str = Field(..., description="The ID of the peer")
+    peer_kind: str = Field(..., description="The node kind of the peer")
+    peer_status: UpdateStatus = Field(
+        ..., description="Indicate how the relationship to this peer was changed in this update"
+    )
+    properties: list[UpdatedProperty] = Field(
+        default_factory=list, description="Changes to properties of this relationship if any were made"
+    )
+
+
+class UpdatedRelationshipCardinalityMany(BaseModel):
+    name: str
+    peers: list[UpdatedRelationshipPeer] = Field(default_factory=list)
+
+    @computed_field
+    def cardinality(self) -> str:
+        return "many"
+
+
+###############################################################################
+# Nodes
+###############################################################################
+
+
+class MutatedNode(BaseModel):
+    node_id: str
+    node_kind: str
+    display_label: str
```

Comment:
`display_label` might be empty and I wonder if we want to display `hfid` if `display_label` is empty?